src/extent.c

```diff
@@ -312,11 +316,17 @@ extents_insert_locked(tsdn_t *tsdn, extents_t *extents, extent_t *extent,
 	size_t size = extent_size_get(extent);
 	size_t psz = extent_size_quantize_floor(size);
 	pszind_t pind = sz_psz2ind(psz);
-	if (extent_heap_empty(&extents->heaps[pind])) {
-		bitmap_unset(extents->bitmap, &extents_bitmap_info,
-		    (size_t)pind);
+	if (config_sized_alloc_region && extent_in_sized_alloc_region(extent)) {
+		szind_t szind = extent_szind_get(extent);
+		extent_heap_insert(&extents->sized_alloc_region_heaps[szind],
+		    extent);
+	} else {
+		if (extent_heap_empty(&extents->other_heaps[pind])) {
+			bitmap_unset(extents->bitmap, &extents_bitmap_info,
```

Comment:
Why is this logic omitted in the ```config_sized_alloc_region``` branch?  The same question applies to ```extents_remove_locked()```.  My guess is that you're keeping slabs in the per bin heaps forever once allocated, even when they're completely empty.  I think this strategy is insufficient in the context of arena reset, because there's no mechanism for reusing the slabs that were in the heaps at reset time.